go/tasks/plugins/hive/executor.go

```diff
@@ -61,6 +61,14 @@ func (q QuboleHiveExecutor) Handle(ctx context.Context, tCtx core.TaskExecutionC
 	// If no error, then infer the new Phase from the various states
 	phaseInfo := MapExecutionStateToPhaseInfo(outgoingState, q.quboleClient)
 
+	if phaseInfo.Phase() == core.PhaseSuccess {
```

Comment:
@EngHabu This uses the barrier transition - this guarantees for the most part that this will only trigger once per successful task right?

Reply:
Yup. Uses the response from `MapExecutionStateToPhaseInfo`

Reply:
yes you are right.

Reply:
@anandswaminathan  should we not do this in the common task layer? and use task_type as the label?

Reply:
Agreed with Ketan, I think it should move higher up in the stack with task_type as a label (or even part of the metric key)